### PR TITLE
PR: Try to decrease timeouts when running tests and check general tests robustness and speed (CI/Testing)

### PR DESCRIPTION
**Pull request by dalthviz**
## Description of Changes

Some changes to improve CI tests runs robustness:

* Check if closing the mainwindow instance for tests marked as `know_leak` helps reducing timeouts/number of tests runs attempts
* Add missing logic to close editorwindows.
* Remove usage of `connect_to_oi` Editor config option
* Add order mark to a couple of test that were failing consistently: `test_add_history` and `test_save_with_preferred_eol_chars`
* Comment out style application over plots and sidebardialog fixtures to decrease chance of segfault over fast tests 
* Prevent `spyder_kernels.comms.commbase.CommError: The comm is not connected.` on Windows when running IPython Console test by setting matplotlib backend to inline when running test instead of trying to get it via `shellwidget.get_matplotlib_backend()`
* Prevent plots `RuntimeError` when testing due to scrollbars timer logic
* Add logic to `test_updates_appenv` to prevent failing due to possible HTTP 403 errors
* Change IPython Console tests percentage being run
* Quit IPdb in some test before finishing them

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: dalthviz

### CODE REVIEW

**Comment by ccordoba12** [comment]
@dalthviz, while you're checking this, you could also decrease the percentage of IPython console tests run in the slow slots on Windows:

https://github.com/spyder-ide/spyder/blob/1fda346d60ceb283fe14c3ffe311e7d4e5f52fd2/conftest.py#L70-L87

That will make those slots run faster.

File: conftest.py
Comment (ccordoba12) [inline conftest.py:87]: Times are still high for the slow slots. Perhaps you should reduce this to `0.2` or `0.1`.

File: conftest.py
Comment (ccordoba12) [inline conftest.py:87]:
```suggestion
            percentage = 0.4
```

Let's change this to `0.4` to see if the slow and fast slots on Windows use about the same time.

File: runtests.py
Comment (ccordoba12) [inline runtests.py:37]:
```suggestion
                   '--timeout_method=thread']
```

This shouldn't be necessary now that this PR is ready for review, right?

File: spyder/app/tests/conftest.py
Comment (ccordoba12) [inline spyder/app/tests/conftest.py:487]:
```suggestion
            # this marker.
            # Also, try to decrease chances of freezes/timeouts from tests that
            # are known to have leaks by also closing the main window for them.
```

File: spyder/plugins/editor/tests/test_plugin.py
Comment (ccordoba12) [inline spyder/plugins/editor/tests/test_plugin.py:419]:
```suggestion
    # Check default options value are set
```

File: spyder/plugins/plots/tests/test_plugin.py
Comment (ccordoba12) [inline spyder/plugins/plots/tests/test_plugin.py:32]:
```suggestion
    if not running_in_ci():
        qapp.setStyleSheet(str(APP_STYLESHEET))
```

- I didn't know doing this was causing segfaults on CIs but it's useful for local testing.
- Please add the missing import.

File: spyder/plugins/plots/widgets/figurebrowser.py
Comment (ccordoba12) [inline spyder/plugins/plots/widgets/figurebrowser.py:488]: Please add a comment on top of this line to explain why the try/except is necessary.

File: spyder/widgets/tests/test_sidebardialog.py
Comment (ccordoba12) [inline spyder/widgets/tests/test_sidebardialog.py:60]:
```suggestion
    if running_in_ci():
        qapp.setStyleSheet(str(APP_STYLESHEET))
```

**Review by ccordoba12 — COMMENTED** [review]
Thanks @dalthviz for your hard work on this!

**Comment by ccordoba12** [comment]
There's also a small error when running the teardown of `test_outline_in_maximized_editor` on the Windows slot tests, but that should be easy to fix.

File: spyder/widgets/tests/test_sidebardialog.py
Comment (ccordoba12) [inline spyder/widgets/tests/test_sidebardialog.py:59]:
```suggestion
    if not running_in_ci():
```

This was my mistake, sorry.

File: spyder/plugins/plots/widgets/figurebrowser.py
Comment (dalthviz) [reply spyder/plugins/plots/widgets/figurebrowser.py]: Checking this again I think the try/except is not necessary so I removed it :+1:

**Review by ccordoba12 — APPROVED** [review]
This is great! Thanks @dalthviz for taking the time to take a look at it.

File: spyder/plugins/updatemanager/tests/test_update_manager.py
Comment (mrclary) [inline spyder/plugins/updatemanager/tests/test_update_manager.py:71]: This should not be an issue now with #22118. @dalthviz, were you still seeing this error after #22118 was merged?

Comment (ccordoba12) [reply spyder/plugins/updatemanager/tests/test_update_manager.py:71]: I think it's better to leave it to prevent unnecessary errors in our test suite.

Comment (dalthviz) [reply spyder/plugins/updatemanager/tests/test_update_manager.py:71]: > @dalthviz, were you still seeing this error after https://github.com/spyder-ide/spyder/pull/22118 was merged?

I think not but since there is a need for the `GITHUB_TOKEN` I though that maybe leaving the workaround could be worthy

Comment (mrclary) [reply spyder/plugins/updatemanager/tests/test_update_manager.py:71]: Thanks @ccordoba12 and @dalthviz. I just wanted to make sure that we weren't still having the rate limit issue after #22118.

**Pull request merged**
